parsers/FR_O.py

```diff
@@ -0,0 +1,277 @@
+}
+# Sources
+HISTORICAL_SOURCE = 'https://opendata-corse-outremer.edf.fr'
+REAL_TIME_SOURCES = {
+    'RE': 'opendata-reunion.edf.fr'
+}
+# Managed zones
+zone_key_mapping = {
+    'FR-COR': 'Corse',
+    'GF': 'Guyane',
+    'GP': 'Guadeloupe',
+    'MQ': 'Martinique',
+    'RE': 'Réunion'
+}
+
+# -------------- Parametrized properties functions -------------- #
+
+
+def get_param(zone_key: None, target_datetime: None):
+    if(target_datetime is None):
+        params = {
+            'RE': {
+                'dataset': 'prod-electricite-temps-reel',
+                'timezone': 'Indian/Reunion',
+                'sort': 'date',
+                'rows': 1
+            }
+        }
+        return params[zone_key]
+    else:
+        datetime = arrow.get(target_datetime, 'Europe/Paris')
+        formatted = datetime.format('YYYY-MM-DDTHH:mm')
+        return {
+            'dataset': 'courbe-de-charge-de-la-production-delectricite-par-filiere',
+            'timezone': 'Europe/Paris',
+            'q': 'date_heure >= {} AND date_heure <= {}'.format(
+                formatted, formatted),
+            'sort': 'date_heure',
+            'rows': 1,
+            'facet': 'territoire',
+            'refine.territoire': zone_key_mapping[zone_key]
+        }
+
+
+def get_api(zone_key: None, target_datetime: None):
+    if(target_datetime is None):
+        return REAL_TIME_APIS[zone_key]
+    else:
+        return HISTORICAL_API_ENDPOINT
+
+
+def get_source(zone_key: None, target_datetime: None):
+    if(target_datetime is None):
+        return REAL_TIME_SOURCES[zone_key]
+    else:
+        return HISTORICAL_SOURCE
+
+
+# -------------- Bagasse / Coal Repartitions -------------- #
+
+# Depending on the month, this correspond more or less to bagasse or coal.
+# This map is an clumsy approximation using harvesting period and annual
+# percentage of biomass used. Here, we use  17.17% for this percentage
+# (https://fr.wikipedia.org/wiki/Usine_de_Bois_Rouge &
+# https://fr.wikipedia.org/wiki/Usine_du_Gol)
+MAP_GENERATION_BAGASSE_COAL_REUNION = {
+    1: {'coal': 1, 'biomass': 0},
+    2: {'coal': 1, 'biomass': 0},
+    3: {'coal': 1, 'biomass': 0},
+    4: {'coal': 1, 'biomass': 0},
+    5: {'coal': 1, 'biomass': 0},
+    6: {'coal': 1, 'biomass': 0},
+    7: {'coal': 0.77, 'biomass': 0.23},
+    8: {'coal': 0.6, 'biomass': 0.4},
+    9: {'coal': 0.6, 'biomass': 0.4},
+    10: {'coal': 0.6, 'biomass': 0.4},
+    11: {'coal': 0.6, 'biomass': 0.4},
+    12: {'coal': 0.77, 'biomass': 0.23}
+}
+# Depending on the month, this correspond more or less to bagasse or coal.
+# This map is an clumsy approximation using harvesting period and annual
+# percentage of biomass used. Here, we use  20% for this percentage
+# (https://fr.wikipedia.org/wiki/Syst%C3%A8me_%C3%A9lectrique_de_la_Guadeloupe &
+# https://www.guadeloupe-energie.gp/wp-content/uploads/2011/07/2010-10-01_Biomasse_Etat-des-lieux.pdf p.35)
+MAP_GENERATION_BAGASSE_COAL_GUADELOUPE = {
```

Comment:
The generation mapping changes month to month due to seasonal use of biomass?

Reply:
Yes, exactly. For Reunion and Guadeloupe, we only have information about biomass/coal power plants. We don't know the repartition. So I'm doing an approximation with the total biomass used in the year and the harvest period.

For the reunion, I made a little study to see if there was a correlation between the real-time biomass input for sugar industry (sugar cane) and the power plant electricity production, but I found no correlation.
I'm not able to attach my documents and scripts (the file can't be processed), so I put screenshots :
![image](https://user-images.githubusercontent.com/6637101/66041092-4ec53180-e519-11e9-9177-27ad85932c75.png)
![image](https://user-images.githubusercontent.com/6637101/66041119-60a6d480-e519-11e9-9914-7f40127d2b69.png)

Reply:
Nice I like it!

Reply:
Yes, but there is a lack of precision, too bad. The source is computed depending on the month and by a gaussian-like distribution of the biomass use.
In the end it's only a monthly multiplicator :/